### PR TITLE
Support frontend schedule scripts

### PR DESCRIPTION
**Pull request by pmconne**
https://github.com/iTwin/itwinjs-backlog/issues/92

Remove the constraint that schedule scripts must be hosted on some element in the iModel. This is feature-gated by `TileAdmin.Props.enableFrontendScheduleScripts`. With that feature enabled, instead of creating specially-made tiles on the backend based on the specific script, we request normal tiles and split their contents on the frontend according to the script.

A forthcoming issue will describe additional optimizations and enhancements to support this feature. For example:
- The work of splitting the tile contents into nodes could be optimized and moved into a web worker.
- The constraint that prohibits instancing when a schedule script is in effect could be lifted, enabling use of the tiles generated and cached by tile agents.

### CODE REVIEW

File: core/frontend/src/render/primitives/VertexTableSplitter.ts [hunk 1]
```diff
@@ -0,0 +1,614 @@
...
+  }
+
+  public ensureCapacity(newCapacity: number): number {
+    if (this.capacity >= newCapacity)
+      return this.capacity;
+
+    newCapacity *= this._growthFactor;
+    const prevData = this._data;
+    this._data = new this._constructor(newCapacity);
+    this._data.set(prevData, 0);
```
Comment (pmconne) [inline core/frontend/src/render/primitives/VertexTableSplitter.ts:53]: @dassaf4 some potential easy improvements for the various growable array types like GrowableFloat64Array: replace loops in ensureCapacity and clone with calls to `set` and consider a growth factor closer to 1.5 than 2.

Comment (dassaf4) [reply core/frontend/src/render/primitives/VertexTableSplitter.ts:53]: Nice! Good idea. Will do this.

File: core/frontend/src/render/primitives/VertexTableSplitter.ts [hunk 2]
```diff
@@ -245,19 +254,98 @@ class ColorTableRemapper {
...
+    exponent = exponent - bias;
+    return base * Math.pow (10.0, exponent);
+  }
+
+  private materialFromAtlasEntry(entry: Uint32Array): SurfaceMaterial | undefined {
+    // eslint-disable-next-line deprecation/deprecation
+    const materialParams = new RenderMaterial.Params();
+    const rgbOverridden = (entry[1] & 1) !== 0;
+    const alphaOverridden = (entry[1] & 2) !== 0;
+    // ###TODO: do we need a key?
```
Comment (pmconne) [inline core/frontend/src/render/primitives/VertexTableSplitter.ts:307]: No, we don't

File: core/frontend/src/render/primitives/VertexTableSplitter.ts [hunk 3]
```diff
@@ -245,19 +254,98 @@ class ColorTableRemapper {
...
+    materialParams.diffuseColor = rgbOverridden ? ColorDef.fromTbgr(entry[0] & 0xffffff) : undefined;
+    materialParams.alpha = alphaOverridden ? (entry[0] >>> 24) / 255.0 : undefined;
+    materialParams.diffuse = (entry[1] >>> 24) / 255.0;
+    materialParams.specular = ((entry[1] >>> 16) & 0xff) / 255.0;
+    materialParams.specularColor = ColorDef.fromTbgr(entry[2]);
+    materialParams.specularExponent = this.unpackFloat (entry[3]);
+
+    // ###TODO: figure out best way to get iModel
```
Comment (pmconne) [inline core/frontend/src/render/primitives/VertexTableSplitter.ts:315]: Use non-deprecated createRenderMaterial which does not require iModel.

**Comment by pmconne** [comment]
@DStradley I will finish this.

**Comment by DStradley** [comment]
> @DStradley I will finish this.

I just posted a fix for materialFromAtlasEntry that I hadn't posted yet.  I didn't bother with the half baked tests I had started.

**Review by DStradley — APPROVED** [review]

**Review by markschlosseratbentley — APPROVED** [review]

**Comment by pmconne** [comment]
@dassaf4 @bbastings or @mgooding please review small set of core-geometry changes.

**Review by mgooding — APPROVED** [review]

**Pull request merged**